lazy_dataset/core.py

```diff
@@ -1664,7 +1689,7 @@ class ListDataset(Dataset):
     """
 
     def __init__(self, examples, name=None):
-        assert isinstance(examples, (tuple, list)), (type(examples), examples)
+        assert isinstance(examples, (tuple, list, NumpySerializedList)), (type(examples), examples)
```

Comment:
Could you change this to test for `collections.UserList`?

In theory, we could also check for `collections.abc.Sequence`, but I am not sure, if it could hide bugs. (e.g. `str` is a sequence and examples shouldn't be a `str`.)